spec/system/user_adds_picture_to_recipe_spec.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'support/features/clearance_helpers'
+
+RSpec.feature 'A signed in user adds a picture to their recipe' do
+  scenario 'after the recipe has been created' do
+    user = create(:user)
+    recipe = create(:recipe, user: user)
+
+    sign_in_with(user.email, user.password)
+    visit '/'
+    click_link I18n.t('layouts.application.view_recipes')
+    click_link I18n.t('recipes.index.show_recipe')
+    click_button I18n.t('recipes.show.add_picture')
+
+    expect(page).to have_current_path(recipe_pictures_path(recipe.id))
```

Comment:
I think this is a good test but I think an even better test would be one that gets all the way through to adding an image. I haven't ever done this (I've only done it with paperclip) so I'm not sure exactly the process. If you're up for reading a bunch, I found these these things:

* https://www.neontsunami.com/posts/testing-activestorage-uploads-in-rails-52
* https://collectiveidea.com/blog/archives/2017/01/16/testing-an-uploaded-file-in-rspec

So some mixture of configuring active storage for a testing environment (which may already be done because this is rails 6) and then reading and uploading a file in the test. There's a lot of detail I'm skipping over here but I don't yet know the answer.

I'd be happy to pair with you on a Friday sometime if you like! Alternatively you can ask questions here if you get stuck.

Reply:
Thanks Edward! I'll do some reading and try to implement this on my own, and follow up with you for that offer :)